### PR TITLE
fix a couple issues with Microsoft.XmlSerailizer.Generator

### PR DESCRIPTION
**Pull request by HongGit**
#53039 - add support for incremental build
#2335 - support add created assembly to Nuget package

### CODE REVIEW

File: src/libraries/Microsoft.XmlSerializer.Generator/src/GenerateNupkgProps.targets [hunk 1]
```diff
@@ -19,6 +19,12 @@
   &lt;ItemGroup&gt;
     &lt;DotNetCliToolReference Include="$(PackageId)" Version="$(PackageVersion)" /&gt;
   &lt;/ItemGroup&gt;
+  &lt;PropertyGroup&gt;
+    &lt;AddSerializers&gt;False&lt;/AddSerializers&gt;
```
Comment (StephenMolloy) [inline src/libraries/Microsoft.XmlSerializer.Generator/src/GenerateNupkgProps.targets:23]: My gut says this should default to True? Doing so is a behavior change, but it's a behavior that people have noticed is missing and have been asking to get fixed. Perhaps others with more experience in this area have more informed opinions than my gut though?

File: src/libraries/Microsoft.XmlSerializer.Generator/src/GenerateNupkgProps.targets [hunk 2]
```diff
@@ -19,6 +19,12 @@
   &lt;ItemGroup&gt;
     &lt;DotNetCliToolReference Include="$(PackageId)" Version="$(PackageVersion)" /&gt;
   &lt;/ItemGroup&gt;
+  &lt;PropertyGroup&gt;
+    &lt;SgenPackSerializer Condition="'%24(SgenPackSerializer)'==''"&gt;False&lt;/SgenPackSerializer&gt;
```
Comment (mconnew) [inline src/libraries/Microsoft.XmlSerializer.Generator/src/GenerateNupkgProps.targets:23]: Should default to true.

File: src/libraries/Microsoft.XmlSerializer.Generator/src/GenerateNupkgProps.targets [hunk 3]
```diff
@@ -19,6 +19,12 @@
...
+  &lt;/PropertyGroup&gt;
+  &lt;ItemGroup Condition="%24(SgenPackSerializer)==True"&gt;
```
Comment (mconnew) [inline src/libraries/Microsoft.XmlSerializer.Generator/src/GenerateNupkgProps.targets:25]: Should use single quotes around both sides of the equals comparison.
```suggestion
  &lt;ItemGroup Condition="'%24(SgenPackSerializer)'=='True'"&gt;
```

File: src/libraries/Microsoft.XmlSerializer.Generator/src/GenerateNupkgProps.targets (hunk 2 above)
Comment (HongGit) [reply src/libraries/Microsoft.XmlSerializer.Generator/src/GenerateNupkgProps.targets:23]: updated

File: src/libraries/Microsoft.XmlSerializer.Generator/src/GenerateNupkgProps.targets (hunk 3 above)
Comment (HongGit) [reply src/libraries/Microsoft.XmlSerializer.Generator/src/GenerateNupkgProps.targets:25]: updated

**Review by StephenMolloy — APPROVED** [review]
lgtm

**Pull request merged**